OpenSim/Common/Exception.h

```diff
@@ -67,14 +91,29 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
 
 //=============================================================================
 // METHODS
 //=============================================================================
 public:
     // CONSTRUCTORS
-    Exception(const std::string &aMsg="",const std::string &aFile="",int aLine=-1);
+    Exception(const std::string &aMsg="",
+              const std::string &aFile="",
+              int aLine=-1);
+
+    /** Call this constructor from derived classes to add file, line and 
+    function information to the error message                                 */
+    Exception(const std::string& file,
```

Comment:
In my comment, I had mentioned the scheme of having derived class constructors call their direct base class with their message. In this scheme, the classes have to remember to delimit the messages using a common delimiter defined in OpenSim::Exception. I chose to move this responsibility to OpenSim::Exception. The purpose of the function addMessage is to have the operation of appending in one place (OpenSim::Exception), allowing it to choose a separator to delimit the messages from different classes in the hierarchy.

@sherm1 -- I don't understand your comment. Apart from when an exception is thrown, under what circumstance would the message be formatted ?